pageserve/templates/base.html

```diff
@@ -32,44 +34,60 @@
 </head>
 
 <body>
+    <div class="topnav">
+        <div id="nav_left">
+            <a class="{% block home_tab_active %}{% endblock %} navtab navbtn" href='/v1/'>Home</a>
+            <a class="{% block events_tab_active %}{% endblock %} navtab navbtn" href='/v1/events'>Events</a>
+        </div>
+        <div id="user_nav">
+            {% block login_buttons %}
+            {% if not session['user'] %}
+            <div id="sign_in_button">
+                <div class="g-signin2" data-onsuccess="onSignIn" data-theme="dark"></div>
+                <script>
+                    function onSignIn(googleUser) {
+                        // Secure Google ID token to send to backend for verification.
+                        var id_token = googleUser.getAuthResponse().id_token;
+                        var xhr = new XMLHttpRequest();
```

Comment:
XHR is probably fine, but there's also the [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) if you want to use promises rather than callbacks.